main-script.cpp

```diff
@@ -0,0 +1,79 @@
+#include <iostream>
+#include "matrix-matrix-product.hpp"
+#include "matrix-vector-product.hpp"
+#include "read-matrix-market.hpp"
+#include "write-matrix-market.hpp"
+
+using namespace std;
+typedef vector<vector<double>> Matrix;
+typedef vector<double> Vector;
+
+int main(int argc, char* argv[]) {
+    const char* file1 = argv[1];
+    const char* file2 = argv[2];
+    int operation_type = atoi(argv[3]);
+
+    if (operation_type == 0) {
+        // Read the matrix and vector
+        Matrix m1 = read_matrix_market(file1); // Read in the matrix file.
+        cout << "Matrix 1 loaded." << endl; 
+
+        Matrix v1 = read_matrix_market(file2); // Read in the vector file.
+        cout << "Vector 1 loaded." << endl;
+	cout << "\n";
+
+	if (v1[0].size() != 1) {
+        cout << "Error: Input matrix is not a single-column vector." << endl;
+        exit(1);
+    	}
+
+    	Vector v1_conv(v1.size()); // define vector to store values from v1
+    	/*
+	 * This loop below takes values from v1 and stores them in a new vector to convert it to a 1D vector
+	 */
```

Comment:
Any time you find yourself writing a comment that says "part 1 does this operation", that tells you to refactor that section of code into a function. The function documentation and function name will make your code more readable.